### PR TITLE
MueLu: Merge Kokkos and non-Kokkos AmalgamationInfo&Factory

### LINKED ISSUE #11719: MueLu: Merge AmalgamationInfo and AmalgamationInfo_kokkos

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu 

## Motivation
Merges Amalgamation / Amalgamation_kokkos and AmalgamationFactory / AmalgamationFactory_kokkos.

Closes #11719

### CODE REVIEW

File: packages/muelu/src/Graph/MatrixTransformation/MueLu_AmalgamationInfo_def.hpp
Comment (GrahamBenHarper) [inline packages/muelu/src/Graph/MatrixTransformation/MueLu_AmalgamationInfo_def.hpp:95]: This seems a little gross to see this duplication in a few places, but I guess it's necessary until we merge `Aggregates` and `Aggregates_kokkos`?

**Review by GrahamBenHarper — APPROVED** [review]
Looks good to me. I assume the `Aggregates` and `Aggregates_kokkos` merge will be one of the last things we do?

**Comment by cgcgcg** [comment]
@GrahamBenHarper  No, I'm just waiting for you to merge Aggregates and Aggregates_kokkos.

File: packages/muelu/src/Graph/MatrixTransformation/MueLu_AmalgamationInfo_def.hpp
Comment (cgcgcg) [reply packages/muelu/src/Graph/MatrixTransformation/MueLu_AmalgamationInfo_def.hpp:95]: I could have made Aggregates a parent class for Aggregates_kokkos, but I figured you'd get rid of it anyway soon.

**Comment by GrahamBenHarper** [comment]
Okay that's fair :stuck_out_tongue: I'll take a look at it now

**Comment by GrahamBenHarper** [comment]
@cgcgcg I think it's fine. I was based off this branch but had already made changes when I saw issues.

**Pull request merged**
